File: app/src/main/java/xyz/yxwzyyk/bandwagoncontrol/db/DBHelper.java
package xyz.yxwzyyk.bandwagoncontrol.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

/**
 * Created by yyk on 12/2/15.
 */
public class DBHelper extends SQLiteOpenHelper {

    private static final String DATABASE_NAME = "list.db";
    private static final int DATABASE_VERSION = 1;

    public DBHelper(Context context){
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        /**
         * 数据结构
         * _id : id
         * title : 标题
         * veid : 主机ID
         * key : 主机密钥
         */
        String sql = "CREATE TABLE IF NOT EXISTS list (_id INTEGER PRIMARY KEY ,title VARCHAR, veid VARCHAR, key VARCHAR)";
        db.execSQL(sql);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }
}

File: app/src/main/java/xyz/yxwzyyk/bandwagoncontrol/db/DBHelper.kt
package xyz.yxwzyyk.bandwagoncontrol.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import org.jetbrains.anko.db.*
import xyz.yxwzyyk.bandwagoncontrol.app.MainApplication

/**
 * Created by yyk on 3/4/16.
 */
class DBHelper(content: Context = MainApplication.instance) : ManagedSQLiteOpenHelper(content, DATABASE_NAME, null, DATABASE_VERSION) {
    companion object {
        val DATABASE_NAME = "host.db"
        val DATABASE_VERSION = 1
        val instance: DBHelper by lazy { DBHelper() }
    }

    override fun onCreate(p0: SQLiteDatabase) {
        p0.createTable(HostTable.NAME, true,
                HostTable.ID to INTEGER + PRIMARY_KEY,
                HostTable.TITLE to TEXT,
                HostTable.VEID to TEXT,
                HostTable.KEY to TEXT)
    }

    override fun onUpgrade(p0: SQLiteDatabase, p1: Int, p2: Int) {
        p0.dropTable(HostTable.NAME, true)
        onCreate(p0)
    }

}

val Context.database: DBHelper
    get() = DBHelper.instance